Composer/packages/client/src/components/NotificationCard.tsx

```diff
@@ -0,0 +1,259 @@
+  retentionTime?: number;
+  link?: ILink;
+  onRenderCardContent?: (props: ICardProps) => JSX.Element;
+}
+
+export interface INotificationProps {
+  id: string;
+  cardProps: ICardProps;
+  onDismiss: (id: string) => void;
+}
+
+export class Timer {
+  timerId: NodeJS.Timeout;
+  start: number;
+  remaining: number;
+  pausing = false;
+  callback: () => void;
+
+  constructor(callback: () => void, delay: number) {
+    this.remaining = delay;
+    this.callback = callback;
+    this.start = Date.now();
+    this.timerId = setTimeout(callback, this.remaining);
+  }
+
+  pause() {
+    if (!this.pausing) {
+      clearTimeout(this.timerId);
+      this.remaining -= Date.now() - this.start;
+      this.pausing = true;
+    }
+  }
+
+  resume() {
+    this.pausing = false;
+    this.start = Date.now();
+    clearTimeout(this.timerId);
+    this.timerId = setTimeout(this.callback, this.remaining);
+  }
+
+  clear() {
+    clearTimeout(this.timerId);
+  }
+}
+
+const renderCardContent = (props: ICardProps) => {
+  const { title, description, type, link } = props;
+  return (
+    <div css={cardContent}>
+      {type === NotificationType.error && <Icon css={errorType} iconName="ErrorBadge" />}
+      {type === NotificationType.success && <Icon css={successType} iconName="Completed" />}
+      <div css={cardDetail}>
+        <div css={cardTitle}>{title}</div>
+        <div css={cardDescription}>{description}</div>
+        {link && (
+          <ActionButton css={linkButton} onClick={link.onClick}>
+            {link.label}
+          </ActionButton>
+        )}
+        {type === NotificationType.loading && (
+          <Shimmer shimmerElements={[{ type: ShimmerElementType.line, height: 2 }]} styles={getShimmerStyles} />
+        )}
+      </div>
+    </div>
+  );
+};
+
+export const NotificationCard = (props: INotificationProps) => {
+  const { cardProps, id, onDismiss } = props;
+  const [show, setShow] = useState(true);
+  const containerRef = useRef(null);
+
+  const removeNotification = () => {
+    setShow(false);
+  };
+
+  // notification will disappear in 5 secs
+  const timer = useRef(cardProps.retentionTime ? new Timer(removeNotification, cardProps.retentionTime) : null).current;
+
+  useEffect(() => {
+    return () => {
+      if (timer) {
+        timer.clear();
+      }
+    };
+  }, []);
+
+  const handleMouseOver = () => {
+    // if mouse over stop the time and record the remaining time
+    if (timer) {
+      timer.pause();
+    }
+  };
+
+  const handleMouseLeave = () => {
+    if (timer) {
+      timer.resume();
+    }
+  };
+
+  const handleAnimationEnd = () => {
+    if (!show) onDismiss(id);
+  };
+
+  let renderCard = renderCardContent;
+  if (cardProps.onRenderCardContent) renderCard = cardProps.onRenderCardContent;
+
+  return (
+    <div
+      ref={containerRef}
+      css={cardContainer(show, containerRef.current)}
+      role="presentation"
+      onAnimationEnd={handleAnimationEnd}
+      onFocus={() => {}}
```

Comment:
?

Reply:
this is a eslint rule onMouseOver must be accompanied by onFocus for accessibility, will have a warn. I add this to pass the rule.